File: ai-resume-portal/frontend/src/types/interview.ts
export type InterviewStatus =
  | 'SCHEDULED'
  | 'CONFIRMED'
  | 'DECLINED'
  | 'COMPLETED'
  | 'CANCELLED'
  | 'RESCHEDULED'

export type InterviewType = 'VIDEO_CALL' | 'PHONE_CALL' | 'IN_PERSON'

export interface Interview {
  id: number
  applicationId: number
  jobId: number
  jobTitle: string
  companyName: string
  employerId: number
  employerName: string
  employerEmail: string
  applicantId: number
  applicantName: string
  applicantEmail: string
  scheduledAt: string
  meetingLink?: string | null
  interviewType: InterviewType
  durationMinutes: number
  message?: string | null
  status: InterviewStatus
  candidateNote?: string | null
  createdAt: string
  updatedAt: string
}

export interface ScheduleInterviewRequest {
  applicationId: number
  scheduledAt: string
  meetingLink?: string
  interviewType: InterviewType
  durationMinutes?: number
  message?: string
}

export interface RescheduleInterviewRequest {
  scheduledAt?: string
  meetingLink?: string
  interviewType?: InterviewType
  durationMinutes?: number
  message?: string
}

File: ai-resume-portal/backend/src/main/java/com/portal/interview/entity/Interview.java
package com.portal.interview.entity;

import com.portal.application.entity.JobApplication;
import com.portal.user.entity.User;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

@Entity
@Table(name = "interviews")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Interview {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "application_id", nullable = false)
    private JobApplication application;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "employer_id", nullable = false)
    private User employer;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "applicant_id", nullable = false)
    private User applicant;

    @Column(name = "scheduled_at", nullable = false)
    private LocalDateTime scheduledAt;

    @Column(name = "meeting_link", length = 500)
    private String meetingLink;

    @Enumerated(EnumType.STRING)
    @Column(name = "interview_type", nullable = false, length = 20)
    private InterviewType interviewType = InterviewType.VIDEO_CALL;

    @Column(name = "duration_minutes", nullable = false)
    private Integer durationMinutes = 60;

    @Column(columnDefinition = "TEXT")
    private String message;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private InterviewStatus status = InterviewStatus.SCHEDULED;

    @Column(name = "candidate_note", columnDefinition = "TEXT")
    private String candidateNote;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
